### ISSUE TITLE
Templateformat .js with passthroughFileCopy no longer works with eleventy >0.3.4

### ISSUE THREAD

**Issue by nielskrijger** [11ty/eleventy#181]
Hello!

Thank you for creating and maintaining this fantastic framework.

We have been using eleventy for a couple of months now and recently tried to update eleventy from 0.3.1 to 0.5.0. In our setup we want to passthrough some `*.js` files as and we noticed no `*.js`-files are being copied with eleventy 0.3.4 or higher.

I replicated the issue here:
https://github.com/nielskrijger/eleventy-js-passthrough

What we were doing was the following in our `.eleventy` config:

```js
module.exports = function(eleventyConfig) {
  return {
    templateFormats: ["css", "njk", "js"],
    passthroughFileCopy: true,
    dir: {
      input: "src",
      output: "build"
    }
  };
};
```

Previously any `*.js` that was found was copied to the output directory (similar like `.css`).

The workaround for this issue is:

```js
module.exports = function(eleventyConfig) {
  // As a workaround uncomment the next line for eleventy 0.3.4 and higher
  eleventyConfig.addPassthroughCopy('src/js');

  return {
    templateFormats: ["css", "njk"],
    passthroughFileCopy: true,
    dir: {
      input: "src",
      output: "build"
    }
  };
};
```

While that makes the build process presumably faster, I wasn't able to find out why the `templateFormat: ['js']` has stopped working in Eleventy 0.3.4 or higher from the release notes alone.

Other templateFormats appear to work correctly (we're actually using quite an extensive list: `['css', 'js', 'jpg', 'svg', 'eot', 'ttf', 'woff', 'woff2', 'txt', 'njk'],`, only `js` no longer appears to work. Arguably we should use `addPassthroughCopy(...)` anyway for most of that ;-)

Regards,
Niels

**Comment by kleinfreund** [comment]
Most of these (e.g. `js`, `jpg`, etc.) aren’t **template formats** (for a list, check out the docs for [Configuration: Template Formats](https://www.11ty.io/docs/config/#template-formats)); hence, they shouldn’t be recognized as template formats.

**However**, this contradicts with [Configuration: Pass-through copy](https://www.11ty.io/docs/copy/):

> Although css is not currently a recognized Eleventy template, Eleventy will now search for any *.css files inside of the input directory and copy them to output (keeping directory structure).

So maybe, you need to do something like below. This won’t match based on file extensions, though. These are paths relative to your project root.

```js
module.exports = function(eleventyConfig) {
  // As a workaround uncomment the next line for eleventy 0.3.4 and higher
  eleventyConfig
    .addPassthroughCopy('src/css')
    .addPassthroughCopy('src/js')
    .addPassthroughCopy('src/img')
    .addPassthroughCopy('src/fonts')
    .addPassthroughCopy('src/static');

  return {
    templateFormats: ["njk"],
    passthroughFileCopy: true,
    dir: {
      input: "src",
      output: "build"
    }
  };
};
```

I think only @zachleat can answer whether what you describe is a regression in the code or in the docs.

**Comment by nielskrijger** [comment]
> So maybe, you need to do something like below. This won’t match based on file extensions, though. These are paths relative to your project root.

That is a good suggestion! It should definitely be the recommended approach for the vast majority of setups.

Why I'm still interested in this issue; our project setup is somewhat complex with multiple themes defined in a `src/themes` directory. Those themes contain custom `.css`, `.njk`, fonts etc; but not `*.js` currently. If we were however to include any `*.js` in a theme directory (for example `src/themes/A/js/custom.js`) being able to passthrough based on file extension does make configuration simpler.

**Comment by zachleat** [comment]
Confirmed! This is due to work for #118.

@kleinfreund is right, the .addPassthroughCopy method is preferable and will be faster for you but this is still a bug!

Relevant docs: https://www.11ty.io/docs/copy/#manual-passthrough-copy-(faster)

**Comment by zachleat** [comment]
This will be fixed in 0.5.1! Thanks for reporting with the EXCELLENT REPO TEST CASE (so good—thank you)
